crates/core/src/surfnet/locker.rs

```diff
@@ -2074,6 +2074,21 @@ impl SurfnetSvmLocker {
             for &account_idx in &ix.accounts {
                 all_required_accounts.insert(message_accounts[account_idx as usize]);
             }
+            // If we still don't have any accounts, it means our instruction doesn't have any accounts.
+            // Some instructions don't need to read/write any accounts, so the account list is empty.
+            // However, we're still using an account to sign, so we add that here.
+            {
+                if all_required_accounts.len() == 0 {
+                    let mut mutable_signers =
+                        mutable_signers.iter().cloned().collect::<IndexSet<_>>();
+                    all_required_accounts.append(&mut mutable_signers);
+                }
+                if all_required_accounts.len() == 0 {
+                    let mut readonly_signers =
+                        readonly_signers.iter().cloned().collect::<IndexSet<_>>();
+                    all_required_accounts.append(&mut readonly_signers);
+                }
```

Comment:
Feels like this snippet is either buggy or something subtile is happening and should be documented?
Is the order `mutable` then `readonly` matters?